New healthcare benchmark: careqa

PabloAgustin opened the pull request:
This PR adds the new CareQA benchmark for healthcare evaluation. 
CareQA originates from the Spanish Specialised Healthcare Training (MIR) exams by the Spanish Ministry of Health. It consists on 5,621 QA pairs in a multiple-choice close-ended version, and 3,730 QA pairs in an open-ended version. 

This PR adds the following healthcare tasks as well: mimic-III (Report Sumarization), mediqa_qa2019 (Open ended QA), meddialog (Question Entailment and Open ended QA), medtext (Open ended QA), mts_dialog (Report Writing), olaph (Medical Facutality). 

Paper: https://arxiv.org/abs/2502.06666

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/EleutherAI/lm-evaluation-harness?pullRequest=2714) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/EleutherAI/lm-evaluation-harness?pullRequest=2714) before we can accept your contribution.<br/><hr/>**PabloAgustin** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/EleutherAI/lm-evaluation-harness?pullRequest=2714) it.</sub>

baberabb commented:
Hi! Thanks for the PR! just some minor points:
1. could you run the pre-commit to fix formatting.
2. remove [task_list.txt](https://github.com/EleutherAI/lm-evaluation-harness/pull/2714/files#diff-0ed88f54684c07e027341c4d88179f2530695a43397ca4cc25202233f069df87)
3. add an entry to `lm_eval/tasks/README.md` adding a brief description like all the other tasks

PabloAgustin commented:
Thanks baberabb!

I added the formatting fixes, as well as the missing readmes, and included all tasks in `lm_eval/tasks/README.md`. Finally, I added an agg metric that was missing.

I had a question: one of the tests failed due to imports (in some of the included metrics, radgraph or hf evaluate I think); how do you handle that internally?

Thanks!

baberabb commented:
> Thanks baberabb!
> 
> I added the formatting fixes, as well as the missing readmes, and included all tasks in `lm_eval/tasks/README.md`. Finally, I added an agg metric that was missing.
> 
> I had a question: one of the tests failed due to imports (in some of the included metrics, radgraph or hf evaluate I think); how do you handle that internally?
> 
> Thanks!

Thanks! looks good

If you wrap the imports in a try except at the top of the file, we can raise the exception as soon as the task is imported rather than after the outputs are generated. For e.g. [here](https://github.com/EleutherAI/lm-evaluation-harness/blob/52df63b7b30da53c481ed9090598d9189fab1d91/lm_eval/tasks/minerva_math/utils.py#L10-L17)

PabloAgustin commented:
Done! Should catch the load metric error now I think.

PabloAgustin commented:
Test failed due to missing pip install bert-score again. Modified the try excepts to catch ImportErrors. Test should be able to catch the exception now.

baberabb commented:
Thanks for the PR!
